### PR TITLE
Fixed #26001 -- Made ModelAdmin.search_fields do an integer lookup for IntegerFields

### PR DESCRIPTION
**Pull request by akki**
What type of test, if any, should be written for this.
Does this require a mention in the docs ?

### CODE REVIEW

File: django/contrib/admin/options.py [hunk 1]
```diff
@@ -884,7 +884,11 @@ def construct_search(field_name):
             if field_name.startswith('^'):
                 return "%s__istartswith" % field_name[1:]
             elif field_name.startswith('='):
-                return "%s__iexact" % field_name[1:]
+                field_name = field_name[1:]
+                # Integers don't care about case-sensitivity.
+                if isinstance(self.model._meta.get_field(field_name), models.IntegerField):
+                    return "%s__exact" % field_name
+                return "%s__iexact" % field_name
```
Comment (charettes) [inline django/contrib/admin/options.py]: It's a shame we register the `__iexact` lookup for all fields. I think `not isintance(..., (models.CharField, models.TextField))` would be more appropriate here.

File: django/contrib/admin/options.py [hunk 2]
```diff
@@ -896,9 +900,17 @@ def construct_search(field_name):
...
-                              for orm_lookup in orm_lookups]
+                or_queries = []
+                for orm_lookup in orm_lookups:
+                    if orm_lookup.endswith('__exact'):
+                        # This is an IntegerField.
+                        try:
+                            bit = int(bit)
+                        except ValueError:
+                            break
+                    or_queries.append(models.Q(**{orm_lookup: bit}))
```
Comment (charettes) [inline django/contrib/admin/options.py]: Are you sure this change is required?

From what I can see the `Exact` lookup will make sure to call `IntegerField.get_prep_value` on its right hand side value.

Comment (akki) [inline django/contrib/admin/options.py]: `IntegerField.get_prep_value` gives an error when I try to search a non-integer field like _"a10"_ from the search field. This was to ensure that.

File: django/contrib/admin/options.py [hunk 3]
```diff
@@ -884,7 +884,12 @@ def construct_search(field_name):
             if field_name.startswith('^'):
                 return "%s__istartswith" % field_name[1:]
             elif field_name.startswith('='):
-                return "%s__iexact" % field_name[1:]
+                field_name = field_name[1:]
+                # Only strings care about case-sensitivity.
+                if isinstance(queryset.model._meta.get_field(field_name),
```
Comment (akki) [inline django/contrib/admin/options.py]: @charettes 
`get_field` is failing to make a reverse lookup for the fields `recommender__titletranslation__text` and `titletranslation__text` on the `Recommendation` model for [this model structure](https://github.com/django/django/blob/master/tests/admin_views/models.py#L385) and is raising the error `Recommendation has no field named 'titletranslation__text'`. How can I resolve it ?

File: django/contrib/admin/options.py [hunk 4]
```diff
@@ -896,8 +901,14 @@ def construct_search(field_name):
             orm_lookups = [construct_search(str(search_field))
                            for search_field in search_fields]
             for bit in search_term.split():
-                or_queries = [models.Q(**{orm_lookup: bit})
-                              for orm_lookup in orm_lookups]
+                or_queries = []
+                for orm_lookup in orm_lookups:
+                    if orm_lookup.endswith('__exact'):
+                        try:
+                            bit = self.model._meta.get_field(orm_lookup[:-7]).get_prep_value(bit)
```
Comment (akki) [inline django/contrib/admin/options.py]: Is is all right to use `get_prep_value` here ?

**Comment by charettes** [comment]
As noted on the ticket I think this should be closed in favor of [#26184](https://code.djangoproject.com/ticket/26184) which propose to solve this issue in a more generic way.

**Comment by timgraham** [comment]
Yes, I agree this seems like it's headed in the wrong direction. Hopefully that other ticket will resolve this or at least make it simpler.

**Pull request closed without merging**